lib/webpacker/source.rb

```diff
@@ -1,38 +1,31 @@
-# Translates a logical reference for a pack source into the final path needed in the HTML.
-# This translation takes into account whether digesting is configured to happen, which it
-# is by default in the production environment (as set via
-# `Rails.configuration.x.webpacker[:digesting] = true`).
+require "webpacker/manifest"
+require "webpacker/configuration"
+
+# Translates a logical reference for a pack source into the final
+# path needed in the HTML using generated manifest.json manifest.
 class Webpacker::Source
-  def initialize(name)
-    @name = name
+  class SourceError < StandardError; end
+
+  def initialize(filename)
+    @filename = filename
   end
 
   def path
-    if config[:dev_server_host].present?
-      "#{config[:dev_server_host]}/#{filename}"
-    elsif config[:digesting]
-      File.join(dist_path, digested_filename)
+    if Rails.env.development? && dev_server_enabled?
+      "http://#{dev_server[:host]}:#{dev_server[:port]}/#{filename}"
     else
-      File.join(dist_path, filename)
+      Webpacker::Manifest.lookup(filename)
     end
   end
 
   private
-    attr_accessor :name
-
-    def config
-      Rails.configuration.x.webpacker
-    end
-
-    def digested_filename
-      Webpacker::Digests.lookup(name)
-    end
+    attr_accessor :filename
 
-    def dist_path
-      config[:packs_dist_path]
+    def dev_server
+      Webpacker::Configuration.dev_server
     end
 
-    def filename
-      "#{name}.js"
+    def dev_server_enabled?
+      ENV["DEV_SERVER_ENABLED"] || dev_server[:enabled]
```

Comment:
None, someone pointed out to have this as a runtime config option, but looks like this is overdone. Seems unnecessary for dev-server.